LLMonFHIR/Study Flow/CurrentStudyManager.swift

```diff
@@ -0,0 +1,97 @@
+//
+// This source file is part of the Stanford LLM on FHIR project
+//
+// SPDX-FileCopyrightText: 2025 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+import Spezi
+import SpeziFoundation
+
+
+@MainActor
+@Observable
+final class CurrentStudyManager: Module, EnvironmentAccessible, Sendable {
+    enum StudyActivationError: Error {
+        case failedParsingQRCodePayload(any Error)
+        case expiredTimestamp
+        /// Another study is already active
+        case alreadyActive
+        /// Unable to find a study for the specified id
+        case unknownStudy
+    }
+    
+    @ObservationIgnored @Application(\.logger) private var logger
+    
+    private(set) var currentSurvey: Survey?
+    
+    
+    func configure() {
+        if let studyId = FeatureFlags.enabledUserStudyId {
+            do {
+                try startStudy(withId: studyId)
+            } catch {
+                logger.error("Unable to start feature-flag-triggered study '\(studyId)'")
+            }
+        }
+    }
+    
+    
+    func handleQRCode(payload payloadString: String) throws(StudyActivationError) {
+        let payload: QRCodePayload
+        do {
+            payload = try QRCodePayload(qrCodePayload: payloadString)
+        } catch {
+            throw .failedParsingQRCodePayload(error)
+        }
+        if let expirationTimestamp = payload.expires, expirationTimestamp < .now {
+            throw .expiredTimestamp
+        }
+        try startStudy(withId: payload.studyId)
+    }
+    
+    
+    func startStudy(withId studyId: String) throws(StudyActivationError) {
+        guard currentSurvey == nil || currentSurvey?.id == studyId else {
+            throw .alreadyActive
+        }
+        guard let study = Survey.withId(studyId) else {
+            throw .unknownStudy
+        }
+        currentSurvey = study
+    }
```

Comment:
If we also inject the system prompt here; we might also need to update the schema here.